### ISSUE TITLE
yahooJSON not working again

### ISSUE THREAD

**Issue by carlosala** [finance-quote/finance-quote#369]
Hey! I'm receiving a 401 Unauthorized when fetching a quote with yahooJSON, is anyone else reproducing?

**Comment by bpschuck** [comment]
I am also getting 401 errors. This was sort of expected after the last fix. The surprise was that I didn't expect it to take this long since the last issue.

**Comment by bender2305** [comment]
I can confirm the issue, Stopped working for me from today on using GNC and Finance::Quote.  I'm getting a 401 (Unauthorized) error / Invalid Crumb / Invalid Cookie Error. Any help would be appreciated.

**Comment by bpschuck** [comment]
Please, no more "me too" messages.
Not sure when this will be addressed. Best bet is to use another source for the time being.

**Comment by achampion** [comment]
Would `v8 chart` work, e.g. `https://query1.finance.yahoo.com/v8/finance/chart/AMZN?metrics=high?&interval=1d&range=1d`?

**Comment by bpschuck** [comment]
@achampion,

> If `v11` doesn't come back, could `v8 chart` be an alternative approach, e.g. `https://query1.finance.yahoo.com/v8/finance/chart/AMZN?metrics=high&interval=1d&range=1d`?

The v8 API was something considered the last time a Yahoo API became inaccessible. I believe the v11 API was chosen because a few more data points were available. In this case it may be the Yahoo API of last resort. 

I haven't had much free time as of late and I am hoping that another community member will step up and create a usable PR. There is also a currency module that uses the v11 API that should be replaced along with YahooJSON.pm. 

Thanks.

Bruce S.

**Comment by ka-patel** [comment]
I can do PR if you like. The latest one at https://github.com/ka-patel/dl_quotes/blob/main/FinanceQuote/YahooJSON.pm works as intended.

**Comment by kamanwu** [comment]
> @bpschuck
> 
> I can do PR if you like. The latest one at https://github.com/ka-patel/dl_quotes/blob/main/FinanceQuote/YahooJSON.pm works as intended and it is based on 1.59 branch.

I use finance-quote since GunCash need it.   Just tested this file.  It does not work for me.  GnuCash report finance-quote not install correctly after using this file.

**Comment by ka-patel** [comment]
You may need to install HTTP::CookieJar::LWP Perl module (I’ve update readme over there) since it is now needed to deal with cookies …

 

If it is installed and not working then can you send me the error you are seeing so that I can reproduce.

 

Thanks.

**Comment by Swatinem** [comment]
Here is the output from GnuCash with upstream `1.59` and with your patch:

```
> LANG=C gnucash-cli --quotes dump yahoo_json AMD.F
Finance::Quote reported a failure for symbol AMD.F: Error retrieving quote for AMD.F. Attempt to fetch the URL https://query2.finance.yahoo.com/v11/finance/quoteSummary/?symbol=AMD.F&modules=price,summaryDetail,defaultKeyStatistics resulted in HTTP response 401 (Unauthorized)
```

```
> LANG=C gnucash-cli --quotes dump yahoo_json AMD.F
Finance::Quote reported a failure for symbol AMD.F: Error retrieving quote for AMD.F. Attempt to fetch the URL https://query2.finance.yahoo.com/v11/finance/quoteSummary/?symbols=AMD.F&crumb={"finance":{"result":null,"error":{"code":"Unauthorized","description":"Invalid Cookie"}}}&modules=price,summaryDetail,defaultKeyStatistics resulted in HTTP response 401 (Unauthorized)
```

Looks like it just puts the Unauthorized JSON error into the `crumb` query string parameter.

**Comment by Swatinem** [comment]
here is a quick hack switching over to the `v8 chart` API, sufficient so I can get quotes for gnucash:

```
> LANG=C gnucash-cli --quotes dump yahoo_json AMD.F   
Finance::Quote fields GnuCash uses:
    symbol: AMD.F           <=== required
      date: 04/12/2024      <=== recommended
  currency: EUR             <=== required
      last: 152.86          <=\ 
       nav:                 <=== one of these
     price:                 <=/ 
```

```diff
--- YahooJSON.pm.1.59	2024-04-13 20:57:49.058142453 +0200
+++ YahooJSON.pm	2024-04-13 21:20:00.580192490 +0200
@@ -35,8 +35,8 @@
 
 our $VERSION = '1.59'; # VERSION
 
-my $YIND_URL_HEAD = 'https://query2.finance.yahoo.com/v11/finance/quoteSummary/?symbol=';
-my $YIND_URL_TAIL = '&modules=price,summaryDetail,defaultKeyStatistics';
+my $YIND_URL_HEAD = 'https://query1.finance.yahoo.com/v8/finance/chart/';
+my $YIND_URL_TAIL = '?interval=1d&range=1d';
 
 sub methods {
     return ( yahoo_json => \&yahoo_json,
@@ -91,7 +91,7 @@
             # Requests for invalid symbols sometimes return 200 with an empty
             # JSON result array
             my $json_data_count
-                = scalar @{ $json_data->{'quoteSummary'}{'result'} };
+                = scalar @{ $json_data->{'chart'}{'result'} };
 
             if ( $json_data_count < 1 ) {
                 $info{ $stocks, "success" } = 0;
@@ -100,32 +100,21 @@
 
             }
             else {
+		my $json_quote = $json_data->{'chart'}{'result'}[0];
+		my $meta = $json_quote->{'meta'};
 
-		my $json_resources_price = $json_data->{'quoteSummary'}{'result'}[0]{'price'};
-		my $json_resources_summaryDetail = $json_data->{'quoteSummary'}{'result'}[0]{'summaryDetail'};
-		my $json_resources_defaultKeyStatistics = $json_data->{'quoteSummary'}{'result'}[0]{'defaultKeyStatistics'};
-
-                # TODO: Check if $json_response_type is "Quote"
-                # before attempting anything else
-                my $json_symbol = $json_resources_price->{'symbol'};
-                #    || $json_resources->{'resource'}{'fields'}{'symbol'};
-                my $json_volume = $json_resources_price->{'regularMarketVolume'}{'raw'};
-                my $json_timestamp =
-                    $json_resources_price->{'regularMarketTime'};
-                my $json_name = $json_resources_price->{'shortName'};
-                my $json_type = $json_resources_price->{'quoteType'};
-                my $json_price =
-                    $json_resources_price->{'regularMarketPrice'}{'raw'};
+		my $json_timestamp =
+                    $meta->{'regularMarketTime'};
 
                 $info{ $stocks, "success" } = 1;
                 $info{ $stocks, "exchange" } =
-                    $json_resources_price->{'exchangeName'};
+                    $meta->{'exchangeName'};
                 $info{ $stocks, "method" } = "yahoo_json";
-                $info{ $stocks, "name" }   = $stocks . ' (' . $json_name . ')';
-                $info{ $stocks, "type" }   = $json_type;
-                $info{ $stocks, "last" }   = $json_price;
-                $info{ $stocks, "currency"} = $json_resources_price->{'currency'};
-                $info{ $stocks, "volume" }   = $json_volume;
+		#$info{ $stocks, "name" }   = $stocks . ' (' . $json_name . ')';
+		#$info{ $stocks, "type" }   = $json_type;
+                $info{ $stocks, "last" }   = $meta->{'regularMarketPrice'};
+                $info{ $stocks, "currency"} = $meta->{'currency'};
+                $info{ $stocks, "volume" }   = $meta->{'regularMarketVolume'};
 
                 # The Yahoo JSON interface returns London prices in GBp (pence) instead of GBP (pounds)
                 # and the Yahoo Base had a hack to convert them to GBP.  In theory all the callers
@@ -160,34 +149,6 @@
                     $info{ $stocks, "currency"} = "ILS";
                 }
 
-            # Add extra fields using names as per yahoo to make it easier
-            #  to switch from yahoo to yahooJSON
-            # Code added by goodvibes
-                {
-                  # turn off warnings in this block to fix bogus
-                  # 'Use of uninitialized value in multiplication' warning
-                  # in Strawberry perl 5.18.2 in Windows
-                  local $^W = 0;
-                  $info{ $stocks, "div_yield" } =
-                    $json_resources_summaryDetail->{'trailingAnnualDividendYield'}{'raw'} * 100;
-                }
-                $info{ $stocks, "eps"} =
-                    $json_resources_defaultKeyStatistics->{'trailingEps'}{'raw'};
-		#    $json_resources_summaryDetail->{'epsTrailingTwelveMonths'};
-                $info{ $stocks, "pe"} = $json_resources_summaryDetail->{'trailingPE'}{'raw'};
-                $info{ $stocks, "year_range"} =
-                    sprintf("%12s - %s",
-                        $json_resources_summaryDetail->{"fiftyTwoWeekLow"}{'raw'},
-                        $json_resources_summaryDetail->{'fiftyTwoWeekHigh'}{'raw'});
-                $info{ $stocks, "open"} =
-                    $json_resources_price->{'regularMarketOpen'}{'raw'};
-                $info{ $stocks, "high"} =
-                    $json_resources_price->{'regularMarketDayHigh'}{'raw'};
-                $info{ $stocks, "low"} =
-                    $json_resources_price->{'regularMarketDayLow'}{'raw'};
-                $info{ $stocks, "close"} =
-                    $json_resources_summaryDetail->{'regularMarketPreviousClose'}{'raw'};
-
                 # MS Windows strftime() does not support %T so use %H:%M:%S
                 #  instead.
                 $my_date =
```

**Comment by ka-patel** [comment]
Your output indicates that it did not get the cookies it needs. 

Is HTTP::CookieJar::LWP module installed? 


------ Original Message ------
Received: Sat, 13 Apr 2024 03:04:16 PM EDT
From: Arpad Borsos ***@***.***>
To: finance-quote/finance-quote ***@***.***>Cc: ka-patel
***@***.***>,  Comment ***@***.***>
Subject: Re: [finance-quote/finance-quote] yahooJSON not working again (Issue
#369)

> Here is the output from GnuCash with upstream `1.59` and with your patch:
> 
> ```
> > LANG=C gnucash-cli --quotes dump yahoo_json AMD.F
> Finance::Quote reported a failure for symbol AMD.F: Error retrieving quote
for AMD.F. Attempt to fetch the URL
https://query2.finance.yahoo.com/v11/finance/quoteSummary/?symbol=AMD.F&modules=price,summaryDetail,defaultKeyStatistics
resulted in HTTP response 401 (Unauthorized)
> ```
> 
> ```
> > LANG=C gnucash-cli --quotes dump yahoo_json AMD.F
> Finance::Quote reported a failure for symbol AMD.F: Error retrieving quote
for AMD.F. Attempt to fetch the URL
https://query2.finance.yahoo.com/v11/finance/quoteSummary/?symbols=AMD.F&crumb={"finance":{"result":null,"error":{"code":"Unauthorized","description":"Invalid
Cookie"}}}&modules=price,summaryDetail,defaultKeyStatistics resulted in HTTP
response 401 (Unauthorized)
> ```
> 
> Looks like it just puts the Unauthorized JSON error into the `crumb` query
string parameter.

**Comment by bpschuck** [comment]
@Swatinem,

> here is a quick hack switching over to the `v8 chart` API, sufficient so I can get quotes for gnucash:

Thanks, but I already have a module using v8 written. Saved since the previous Yahoo call (v7 I believe) stopped working in June or July of last year. Easy enough to update it with any mods since then.

@ka-patel,

Thanks again. I'm going to play around and experiment using just `HTTP::Cookies`. Although, `HTTP::CookieJar` is already used in Stooq.pm and Bloomberg.pm.

Bruce S

**Comment by ka-patel** [comment]
I can concur that 'v8 chart' works ... I had added method called YahooChart to
my fork at https://github.com/ka-patel/finance-quote which fetches using 'v8
chart' method.

------ Original Message ------
Received: Sat, 13 Apr 2024 03:24:50 PM EDT
From: Arpad Borsos ***@***.***>
To: finance-quote/finance-quote ***@***.***>Cc: ka-patel
***@***.***>,  Comment ***@***.***>
Subject: Re: [finance-quote/finance-quote] yahooJSON not working again (Issue
#369)

> here is a quick hack switching over to the `v8 chart` API, sufficient so I
can get quotes for gnucash:
> 
> ```
> > LANG=C gnucash-cli --quotes dump yahoo_json AMD.F   
> Finance::Quote fields GnuCash uses:
>     symbol: AMD.F           <=== required
>       date: 04/12/2024      <=== recommended
>   currency: EUR             <=== required
>       last: 152.86          <=\ 
>        nav:                 <=== one of these
>      price:                 <=/ 
> ```
> 
> ```diff
> --- YahooJSON.pm.1.59	2024-04-13 20:57:49.058142453 +0200
> +++ YahooJSON.pm	2024-04-13 21:20:00.580192490 +0200
> @@ -35,8 +35,8 @@
>  
>  our $VERSION = '1.59'; # VERSION
>  
> -my $YIND_URL_HEAD =
'https://query2.finance.yahoo.com/v11/finance/quoteSummary/?symbol=';
> -my $YIND_URL_TAIL = '&modules=price,summaryDetail,defaultKeyStatistics';
> +my $YIND_URL_HEAD = 'https://query1.finance.yahoo.com/v8/finance/chart/';
> +my $YIND_URL_TAIL = '?interval=1d&range=1d';
>  
>  sub methods {
>      return ( yahoo_json => \&yahoo_json,
> @@ -91,7 +91,7 @@
>              # Requests for invalid symbols sometimes return 200 with an
empty
>              # JSON result array
>              my $json_data_count
> -                = scalar @{ $json_data->{'quoteSummary'}{'result'} };
> +                = scalar @{ $json_data->{'chart'}{'result'} };
>  
>              if ( $json_data_count < 1 ) {
>                  $info{ $stocks, "success" } = 0;
> @@ -100,32 +100,21 @@
>  
>              }
>              else {
> +		my $json_quote = $json_data->{'chart'}{'result'}[0];
> +		my $meta = $json_quote->{'meta'};
>  
> -		my $json_resources_price =
$json_data->{'quoteSummary'}{'result'}[0]{'price'};
> -		my $json_resources_summaryDetail =
$json_data->{'quoteSummary'}{'result'}[0]{'summaryDetail'};
> -		my $json_resources_defaultKeyStatistics =
$json_data->{'quoteSummary'}{'result'}[0]{'defaultKeyStatistics'};
> -
> -                # TODO: Check if $json_response_type is "Quote"
> -                # before attempting anything else
> -                my $json_symbol = $json_resources_price->{'symbol'};
> -                #    || $json_resources->{'resource'}{'fields'}{'symbol'};
> -                my $json_volume =
$json_resources_price->{'regularMarketVolume'}{'raw'};
> -                my $json_timestamp =
> -                    $json_resources_price->{'regularMarketTime'};
> -                my $json_name = $json_resources_price->{'shortName'};
> -                my $json_type = $json_resources_price->{'quoteType'};
> -                my $json_price =
> -                    $json_resources_price->{'regularMarketPrice'}{'raw'};
> +		my $json_timestamp =
> +                    $meta->{'regularMarketTime'};
>  
>                  $info{ $stocks, "success" } = 1;
>                  $info{ $stocks, "exchange" } =
> -                    $json_resources_price->{'exchangeName'};
> +                    $meta->{'exchangeName'};
>                  $info{ $stocks, "method" } = "yahoo_json";
> -                $info{ $stocks, "name" }   = $stocks . ' (' . $json_name .
')';
> -                $info{ $stocks, "type" }   = $json_type;
> -                $info{ $stocks, "last" }   = $json_price;
> -                $info{ $stocks, "currency"} =
$json_resources_price->{'currency'};
> -                $info{ $stocks, "volume" }   = $json_volume;
> +		#$info{ $stocks, "name" }   = $stocks . ' (' . $json_name . ')';
> +		#$info{ $stocks, "type" }   = $json_type;
> +                $info{ $stocks, "last" }   =
$meta->{'regularMarketPrice'};
> +                $info{ $stocks, "currency"} = $meta->{'currency'};
> +                $info{ $stocks, "volume" }   =
$meta->{'regularMarketVolume'};
>  
>                  # The Yahoo JSON interface returns London prices in GBp
(pence) instead of GBP (pounds)
>                  # and the Yahoo Base had a hack to convert them to GBP.  In
theory all the callers
> @@ -160,34 +149,6 @@
>                      $info{ $stocks, "currency"} = "ILS";
>                  }
>  
> -            # Add extra fields using names as per yahoo to make it easier
> -            #  to switch from yahoo to yahooJSON
> -            # Code added by goodvibes
> -                {
> -                  # turn off warnings in this block to fix bogus
> -                  # 'Use of uninitialized value in multiplication' warning
> -                  # in Strawberry perl 5.18.2 in Windows
> -                  local $^W = 0;
> -                  $info{ $stocks, "div_yield" } =
> -                   
$json_resources_summaryDetail->{'trailingAnnualDividendYield'}{'raw'} * 100;
> -                }
> -                $info{ $stocks, "eps"} =
> -                   
$json_resources_defaultKeyStatistics->{'trailingEps'}{'raw'};
> -		#    $json_resources_summaryDetail->{'epsTrailingTwelveMonths'};
> -                $info{ $stocks, "pe"} =
$json_resources_summaryDetail->{'trailingPE'}{'raw'};
> -                $info{ $stocks, "year_range"} =
> -                    sprintf("%12s - %s",
> -                       
$json_resources_summaryDetail->{"fiftyTwoWeekLow"}{'raw'},
> -                       
$json_resources_summaryDetail->{'fiftyTwoWeekHigh'}{'raw'});
> -                $info{ $stocks, "open"} =
> -                    $json_resources_price->{'regularMarketOpen'}{'raw'};
> -                $info{ $stocks, "high"} =
> -                   
$json_resources_price->{'regularMarketDayHigh'}{'raw'};
> -                $info{ $stocks, "low"} =
> -                    $json_resources_price->{'regularMarketDayLow'}{'raw'};
> -                $info{ $stocks, "close"} =
> -                   
$json_resources_summaryDetail->{'regularMarketPreviousClose'}{'raw'};
> -
>                  # MS Windows strftime() does not support %T so use
%H:%M:%S

**Comment by Adriano-Baldi** [comment]
> here is a quick hack switching over to the `v8 chart` API, sufficient so I can get quotes for gnucash:
> 
> ```
> > LANG=C gnucash-cli --quotes dump yahoo_json AMD.F   
> Finance::Quote fields GnuCash uses:
>     symbol: AMD.F           <=== required
>       date: 04/12/2024      <=== recommended
>   currency: EUR             <=== required
>       last: 152.86          <=\ 
>        nav:                 <=== one of these
>      price:                 <=/ 
> ```
> 
> ```diff
> --- YahooJSON.pm.1.59	2024-04-13 20:57:49.058142453 +0200
> +++ YahooJSON.pm	2024-04-13 21:20:00.580192490 +0200
> @@ -35,8 +35,8 @@
>  
>  our $VERSION = '1.59'; # VERSION
>  
> -my $YIND_URL_HEAD = 'https://query2.finance.yahoo.com/v11/finance/quoteSummary/?symbol=';
> -my $YIND_URL_TAIL = '&modules=price,summaryDetail,defaultKeyStatistics';
> +my $YIND_URL_HEAD = 'https://query1.finance.yahoo.com/v8/finance/chart/';
> +my $YIND_URL_TAIL = '?interval=1d&range=1d';
>  
>  sub methods {
>      return ( yahoo_json => \&yahoo_json,
> @@ -91,7 +91,7 @@
>              # Requests for invalid symbols sometimes return 200 with an empty
>              # JSON result array
>              my $json_data_count
> -                = scalar @{ $json_data->{'quoteSummary'}{'result'} };
> +                = scalar @{ $json_data->{'chart'}{'result'} };
>  
>              if ( $json_data_count < 1 ) {
>                  $info{ $stocks, "success" } = 0;
> @@ -100,32 +100,21 @@
>  
>              }
>              else {
> +		my $json_quote = $json_data->{'chart'}{'result'}[0];
> +		my $meta = $json_quote->{'meta'};
>  
> -		my $json_resources_price = $json_data->{'quoteSummary'}{'result'}[0]{'price'};
> -		my $json_resources_summaryDetail = $json_data->{'quoteSummary'}{'result'}[0]{'summaryDetail'};
> -		my $json_resources_defaultKeyStatistics = $json_data->{'quoteSummary'}{'result'}[0]{'defaultKeyStatistics'};
> -
> -                # TODO: Check if $json_response_type is "Quote"
> -                # before attempting anything else
> -                my $json_symbol = $json_resources_price->{'symbol'};
> -                #    || $json_resources->{'resource'}{'fields'}{'symbol'};
> -                my $json_volume = $json_resources_price->{'regularMarketVolume'}{'raw'};
> -                my $json_timestamp =
> -                    $json_resources_price->{'regularMarketTime'};
> -                my $json_name = $json_resources_price->{'shortName'};
> -                my $json_type = $json_resources_price->{'quoteType'};
> -                my $json_price =
> -                    $json_resources_price->{'regularMarketPrice'}{'raw'};
> +		my $json_timestamp =
> +                    $meta->{'regularMarketTime'};
>  
>                  $info{ $stocks, "success" } = 1;
>                  $info{ $stocks, "exchange" } =
> -                    $json_resources_price->{'exchangeName'};
> +                    $meta->{'exchangeName'};
>                  $info{ $stocks, "method" } = "yahoo_json";
> -                $info{ $stocks, "name" }   = $stocks . ' (' . $json_name . ')';
> -                $info{ $stocks, "type" }   = $json_type;
> -                $info{ $stocks, "last" }   = $json_price;
> -                $info{ $stocks, "currency"} = $json_resources_price->{'currency'};
> -                $info{ $stocks, "volume" }   = $json_volume;
> +		#$info{ $stocks, "name" }   = $stocks . ' (' . $json_name . ')';
> +		#$info{ $stocks, "type" }   = $json_type;
> +                $info{ $stocks, "last" }   = $meta->{'regularMarketPrice'};
> +                $info{ $stocks, "currency"} = $meta->{'currency'};
> +                $info{ $stocks, "volume" }   = $meta->{'regularMarketVolume'};
>  
>                  # The Yahoo JSON interface returns London prices in GBp (pence) instead of GBP (pounds)
>                  # and the Yahoo Base had a hack to convert them to GBP.  In theory all the callers
> @@ -160,34 +149,6 @@
>                      $info{ $stocks, "currency"} = "ILS";
>                  }
>  
> -            # Add extra fields using names as per yahoo to make it easier
> -            #  to switch from yahoo to yahooJSON
> -            # Code added by goodvibes
> -                {
> -                  # turn off warnings in this block to fix bogus
> -                  # 'Use of uninitialized value in multiplication' warning
> -                  # in Strawberry perl 5.18.2 in Windows
> -                  local $^W = 0;
> -                  $info{ $stocks, "div_yield" } =
> -                    $json_resources_summaryDetail->{'trailingAnnualDividendYield'}{'raw'} * 100;
> -                }
> -                $info{ $stocks, "eps"} =
> -                    $json_resources_defaultKeyStatistics->{'trailingEps'}{'raw'};
> -		#    $json_resources_summaryDetail->{'epsTrailingTwelveMonths'};
> -                $info{ $stocks, "pe"} = $json_resources_summaryDetail->{'trailingPE'}{'raw'};
> -                $info{ $stocks, "year_range"} =
> -                    sprintf("%12s - %s",
> -                        $json_resources_summaryDetail->{"fiftyTwoWeekLow"}{'raw'},
> -                        $json_resources_summaryDetail->{'fiftyTwoWeekHigh'}{'raw'});
> -                $info{ $stocks, "open"} =
> -                    $json_resources_price->{'regularMarketOpen'}{'raw'};
> -                $info{ $stocks, "high"} =
> -                    $json_resources_price->{'regularMarketDayHigh'}{'raw'};
> -                $info{ $stocks, "low"} =
> -                    $json_resources_price->{'regularMarketDayLow'}{'raw'};
> -                $info{ $stocks, "close"} =
> -                    $json_resources_summaryDetail->{'regularMarketPreviousClose'}{'raw'};
> -
>                  # MS Windows strftime() does not support %T so use %H:%M:%S
>                  #  instead.
>                  $my_date =
> ```

Good job.
I confirm that it works for me too.
Thank you

┌──(adriano@Saturn)-[~]
└─$flatpak run --command=sh org.gnucash.GnuCash
[📦 org.gnucash.GnuCash ~]$ gnucash-cli -Q dump yahoo_json BTC-EUR
Finance::Quote fields GnuCash uses:
    symbol: BTC-EUR         <=== required
      date: 04/13/2024      <=== recommended
  currency: EUR             <=== required
      last: 58101.49        <=\ 
       nav:                 <=== one of these
     price:                 <=/ 

[📦 org.gnucash.GnuCash ~]$

**Comment by bpschuck** [comment]
As I've now mentioned 3 times, I already have a v8 based module prepared.

Locking this conversation for the time being.

**Comment by bpschuck** [comment]
Just an FYI, I have worked on getting a PR ready using @ka-patel's changes. I tried applying the same logic to the `CurrencyRates/YahooJSON.pm` currency module with no luck. In that case I quickly modified the currency module to utilize the "v8" API. I admit, I suck at using the developer tool/console in Chrome and Firefox to figure out the magic sauce needed.

There is some miscellaneous housekeeping and cleanup that needs to be done before pushing a final release to CPAN. Plus I had some other changes that I was planning for v1.60 but haven't done yet. Instead of v1.60, I may upload v1.59_01 tomorrow so people can at least fetch the fix for YahooJSON before v1.60 is uploaded and indexed by CPAN.

Bruce S.
